addons/mrp/models/mrp_bom.py

```diff
@@ -202,22 +201,12 @@ def onchange_bom_structure(self):
                 }
             }
 
-    @api.onchange('product_uom_id')
-    def onchange_product_uom_id(self):
-        res = {}
-        if not self.product_uom_id or not self.product_tmpl_id:
-            return
-        if self.product_uom_id.category_id.id != self.product_tmpl_id.uom_id.category_id.id:
-            self.product_uom_id = self.product_tmpl_id.uom_id.id
-            res['warning'] = {'title': _('Warning'), 'message': _('The Product Unit of Measure you chose has a different category than in the product form.')}
-        return res
```

Comment:
Especially for manufacturing, I think a warning would be fair if there was no link (so no real conversion) made between the uom set on the bom and the uom set on the product.